src/access/wac.ts

```diff
@@ -235,3 +264,52 @@ export function getGroupAccessAll(
 ): Promise<AgentWacAccess | null> {
   return getActorAccessAll(resource, getGroupAccessAllWac, options);
 }
+
+/**
+ * Set the Access modes for a given Agent to a given Resource.
+ *
+ * Important note: if the target resource did not have a Resource ACL, and its
+ * Access was regulated by its Fallback ACL, said Fallback ACL is copied to create
+ * a new Resource ACL. This has the side effect that the next time the Fallback
+ * ACL is updated, the changes WILL NOT IMPACT the target resource.
+ *
+ * @param resource The Resource for which Access is being set
+ * @param agent The Agent for whom Access is being set
+ * @param access The Access being set
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns The Resource, with its ACL updated, or null if the new Access could not
+ * be set.
+ */
+export async function setAgentResourceAccess<T extends WithServerResourceInfo>(
+  resource: T,
+  agent: WebId,
+  access: WacAccess,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): Promise<(T & WithResourceAcl) | null> {
+  if (!hasAccessibleAcl(resource)) {
+    return null;
+  }
+  const acl = await internal_fetchAcl(resource, options);
+  const resourceWithAcl = internal_setAcl(resource, acl);
+  let resourceAcl: AclDataset;
+  if (hasResourceAcl(resourceWithAcl)) {
+    // This is the simple case, where the Resource ACL we need to update already
+    // exists.
+    resourceAcl = internal_getResourceAcl(resourceWithAcl);
+  } else if (hasFallbackAcl(resourceWithAcl)) {
+    // In this case, the Resource ACL needs to be created first, and then updated.
+    resourceAcl = createAclFromFallbackAcl(resourceWithAcl);
+  } else {
+    // This probably isn't possible, because otherwise hasAccessibleAcl is false
+    // and the function already returned, so probably exclude from coverage.
+    return null;
+  }
+  const updatedResourceAcl = setAgentResourceAccessWac(
+    resourceAcl,
+    agent,
+    universalAccessToAcl(access)
+  );
+  return internal_setResourceAcl(resourceWithAcl, updatedResourceAcl);
```

Comment:
For some reason, I assumed the ACL would be saved when saving the resource, which is absolutely not the case. I fixed it with ab7ef22a653da203fac5beb3ff279bce166e99ac.

However, this got me thinking: would it be possible to only fetch the ACL if the provided resource didn't have one attached locally, and to only save the ACL when the resource is saved ? I guess it would add risk for conflict, or prevent one from realizing the user doesn't have access to the ACL, so I guess the current approach of always fetching and writing the ACL back is safer.

Reply:
Hmm, only fetching it if it didn't have one sounds like a good idea and doable - we can just do a `hasResourceAcl` and `hasFallbackAcl` before fetching it, I'd presume?

Saving it is more problematic as long as we don't have atomic updates (which I'm starting to think we might have optimised for too early in our API design, given that it doesn't seem like that's actually coming any time soon, and that it makes things like this a bit awkward), because it's not clear what to do if one of the two updates failed.